src/com/google/cose/Ec2SigningKey.java

```diff
@@ -140,7 +140,10 @@ private static byte[] arrayFromBigNum(BigInteger num, int keySize)
     return keyBytes;
   }
 
-  /** Generates a COSE formatted Ec2 signing key given a specific algorithm */
+  /**
+   * Generates a COSE formatted Ec2 signing key given a specific algorithm. The selected hash
+   * algorithm is chosen based from section 6.2.1 of RFC 5656
```

Comment:
The hash algorithm is in the `Algorithm` enum. I think you meant to say `"The selected key size is chosen based on..."`

Reply:
Fixed to read "The selected key size is chosen based on...", done.